src/ToString.php

```diff
@@ -18,6 +19,12 @@ final class ToString implements FilterInterface
      */
     public function filter(mixed $value): mixed
     {
+        if (is_iterable($value)) {
+            return ScalarOrArrayFilterCallback::applyRecursively(
+                $value,
+                $this->filter(...)
+            );
+        }
```

Comment:
```suggestion
        return ScalarOrArrayFilterCallback::applyRecursively(
            $value,
            static fn (string $value): string => $value,
        );
```

Casting is applied inside `applyRecursively`, so this filter literaly has nothing to do.